### ISSUE TITLE
Fix comparison: don't use 'is' for string comparison

### ISSUE THREAD

**Issue by dvolgyes** [getkeops/keops#109]
Hi,

I opened a pull request (PR #108 ), but i also open a ticket because i don't see activity on the PR.
So the python code has a bug: some strings are compared as 'variablename is "stringvalue"'
which is an error. 
Explanation: "is" and "is not" compare addresses of objects, while == and != compare values of objects.
The problem is that some immutable objects are cached/singletons, but some are not.
There is only one "None" object, anything compared to None should be "is None" or "is not None".
But strings could have multiple copies, but not necessarily, and the same is true for integers.

https://stackoverflow.com/questions/1504717/why-does-comparing-strings-using-either-or-is-sometimes-produce-a-differe

Long story short: the is not "auto" and the is "float32" could work, but not guaranteed, and should be replaced with != and ==.

(For future: I recommend to add flake8 plugins to pre-commit hooks, so similar issues could be detected before they are committed. See: https://flake8.pycqa.org/en/latest/user/using-hooks.html  
Pre-commit is really great, you should try it.)

**Comment by bcharlier** [comment]
Hi @dvolgyes ,

thanks for the PR! I will look at pre-commit, thank again for the link.


b.
